src/main/java/com/zaxxer/nuprocess/linux/LinuxProcess.java

```diff
@@ -75,9 +75,10 @@ public NuProcess start(List<String> command, String[] environment, Path cwd) {
 
          afterStart();
 
+         callStart();
```

Comment:
Disclaimer: I know nothing about NuProcess internals, just experiencing the same issue that you do.

As I can see, the document for `onStart` states:

> Note that this method is called <em>at some point after</em> the process

while for `onPreStart` it says:

> Unlike the {@link #onStart(NuProcess)} method, this method is invoked before the process is spawned

Does the statement for `onStart` still hold true?

Reply:
@gagarski onPreStart is called before the actual vfork+exec native call is made. The onStart is called shortly after the native vfork+exec call returns. This is accurate.